commands/bosh_scan.go

```diff
@@ -58,33 +74,47 @@ func (command *BoshScanCommand) Execute(args []string) error {
 	}
 
 	m := sync.Mutex{}
+
 	wg := &sync.WaitGroup{}
 	wg.Add(len(deployments))
-	for _, d := range deployments {
-		logger.Debugf("About to launch go func: %s", d.Name())
-		go func(dep bosh.TargetDeployment) {
-			defer wg.Done()
 
-			logger.Debugf("About to scan: %s", dep.Name())
-			results, err := scanner.Bosh(dep).Scan(&command.FileRegexes, logger)
-			if err != nil {
-				log.Fatalf("failed to scan: %s", err.Error())
-			}
+	results := make(chan ScanResult)
```

Comment:
Make this a channel with size `len(deployments)`. Then, writing to the channel on line 41-43 doesn't need to be in a goroutine.